src/main/java/de/hysky/skyblocker/skyblock/chocolatefactory/ChocolateFactorySolver.java

```diff
@@ -0,0 +1,165 @@
+import java.util.stream.Collectors;
+
+public class ChocolateFactorySolver extends ContainerSolver {
+	private static final Pattern CPS_PATTERN = Pattern.compile("([\\d,.]+) Chocolate per second");
+	private static final Pattern CPS_INCREASE_PATTERN = Pattern.compile("\\+([\\d,]+) Chocolate per second");
+	private static final Pattern COST_PATTERN = Pattern.compile("Cost ([\\d,]+) Chocolate");
+	private static final Pattern TOTAL_MULTIPLIER_PATTERN = Pattern.compile("Total Multiplier: ([\\d.]+)x");
+	private static final Pattern MULTIPLIER_INCREASE_PATTERN = Pattern.compile("\\+([\\d.]+)x Chocolate per second");
+	private static final Pattern TOTAL_CHOCOLATE_PATTERN = Pattern.compile("([\\d,]+) Chocolate");
+
+	public ChocolateFactorySolver() {
+		super("^Chocolate Factory$");
+	}
+
+	@Override
+	protected boolean isEnabled() {
+		return SkyblockerConfigManager.get().helpers.chocolateFactory.enableChocolateFactoryHelper;
+	}
+
+	@Override
+	protected List<ColorHighlight> getColors(String[] groups, Int2ObjectMap<ItemStack> slots) {
+		final Int2ObjectMap<Rabbit> cpsIncreaseFactors = new Int2ObjectLinkedOpenHashMap<>(6);
+		for (int i = 29; i <= 33; i++) { // The 5 rabbits slots are in 29, 30, 31, 32 and 33.
+			ItemStack item = slots.get(i);
+			if (item.getItem() != Items.PLAYER_HEAD || item.isEmpty()) continue;
+
+			int finalI = i; //Java, pfft.
+			getRabbit(item).ifPresent(rabbit -> cpsIncreaseFactors.put(finalI, rabbit));
+		}
+
+		//Coach is in slot 42 while the factory info item is in slot 45.
+		getCoach(slots.get(45), slots.get(42)).ifPresent(coach -> cpsIncreaseFactors.put(42, coach));
+		if (cpsIncreaseFactors.isEmpty()) return List.of(); //Something went wrong.
+
+		OptionalLong totalChocolate = getTotalChocolate(slots.get(13));
+
+		List<Int2ObjectMap.Entry<Rabbit>> sorted = cpsIncreaseFactors.int2ObjectEntrySet()
+		                                                             .stream() //Compare cost/cpsIncrease rather than cpsIncrease/cost to avoid getting close to 0 and losing precision.
+		                                                             .sorted(Comparator.comparingDouble(entry -> entry.getValue().cost() / entry.getValue().cpsIncrease())) //Ascending order, lower = better
+		                                                             .dropWhile(entry -> entry.getValue().cost == 0)
+		                                                             .collect(Collectors.toCollection(LinkedList::new));
+
+		Int2ObjectMap.Entry<Rabbit> bestEntry = sorted.removeFirst();
+		if (totalChocolate.isEmpty() || bestEntry.getValue().cost < totalChocolate.getAsLong()) return List.of(ColorHighlight.green(bestEntry.getIntKey()));
+
+		for (Int2ObjectMap.Entry<Rabbit> entry : sorted) {
+			if (entry.getValue().cost > totalChocolate.getAsLong()) continue;
+			return List.of(ColorHighlight.green(entry.getIntKey()), ColorHighlight.yellow(bestEntry.getIntKey()));
+		}
+		return List.of(ColorHighlight.green(bestEntry.getIntKey()));
```

Comment:
```suggestion
		if (totalChocolate.isEmpty()) return List.of(ColorHighlight.yellow(bestEntry.getIntKey()));

		if (bestEntry.getValue().cost <= totalChocolate.getAsLong()) return List.of(ColorHighlight.green(bestEntry.getIntKey()));

		for (Int2ObjectMap.Entry<Rabbit> entry : sorted) if (entry.getValue().cost <= totalChocolate.getAsLong()) return List.of(ColorHighlight.green(entry.getIntKey()), ColorHighlight.yellow(bestEntry.getIntKey()));

		return List.of(ColorHighlight.yellow(bestEntry.getIntKey()));
```
fix the highlighting logic